DocumentationAnalyzers/DocumentationAnalyzers/Helpers/SymbolExtensions.cs

```diff
@@ -40,18 +39,18 @@ public static bool HasAnyTypeParameter(this ISymbol symbol, string name, StringC
         {
             for (var currentSymbol = symbol; currentSymbol != null; currentSymbol = currentSymbol.ContainingSymbol)
             {
-                switch (currentSymbol.Kind)
+                switch (currentSymbol)
                 {
-                case SymbolKind.NamedType:
-                    if (((INamedTypeSymbol)currentSymbol).TypeParameters.Any(parameter => comparer.Equals(parameter.Name, name)))
+                case INamedTypeSymbol namedType:
```

Comment:
why not `when`

Reply:
I can see how some people would prefer to use `when` here for succinctness. For no particularly good reason other than "I like things I like", I have a preference towards the more verbose form here. I would likely change my mind if the pattern needed to be expanded substantially.

Reply:
yes, it is fine, was mostly curious